From 69dc0a799ace1596d5a5f079db92684907b63882 Mon Sep 17 00:00:00 2001
From: Kevin Chung <kchung@ctfd.io>
Date: Thu, 23 May 2024 18:38:14 -0400
Subject: [PATCH] Improve caching by returning proper Cache-Control header for
 static file handlers (#2543)

* Return a proper Cache-Control header with an hour timeout instead of relying on the conditional browser checks
---
 CTFd/plugins/__init__.py | 2 +-
 CTFd/views.py            | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/CTFd/plugins/__init__.py b/CTFd/plugins/__init__.py
index b8a844b7..93094dc2 100644
--- a/CTFd/plugins/__init__.py
+++ b/CTFd/plugins/__init__.py
@@ -55,7 +55,7 @@ def register_plugin_asset(app, asset_path, admins_only=False, endpoint=None):
         endpoint = asset_path.replace("/", ".")
 
     def asset_handler():
-        return send_file(asset_path)
+        return send_file(asset_path, max_age=3600)
 
     if admins_only:
         asset_handler = admins_only_wrapper(asset_handler)
diff --git a/CTFd/views.py b/CTFd/views.py
index 703b09c6..ed9770b8 100644
--- a/CTFd/views.py
+++ b/CTFd/views.py
@@ -521,7 +521,7 @@ def themes(theme, path):
         if cand_path is None:
             abort(404)
         if os.path.isfile(cand_path):
-            return send_file(cand_path)
+            return send_file(cand_path, max_age=3600)
     abort(404)
 
 
@@ -544,7 +544,7 @@ def themes_beta(theme, path):
         if cand_path is None:
             abort(404)
         if os.path.isfile(cand_path):
-            return send_file(cand_path)
+            return send_file(cand_path, max_age=3600)
     abort(404)
 
 
-- 
2.43.0

